### PR TITLE
8335820: java/lang/invoke/LFCaching/LFSingleThreadCachingTest.java fails due to IllegalArgumentException: hash must be nonzero

### PR DESCRIPTION
**Pull request by asotona**
Class-File API constant pool implementation requires non-zero entry hash code.
Unfortunately current implementation computes zero hash code for specific CP entries.

This patch removes invalid and obsolete `AbstractPoolEntry::phiMix` calculation and assures all pool entries have non-zero hash. A regression test of the actual zero-hash `IntegerEntry` has been added. 

All pre-computed hash codes in `BoundAttribute::standardAttribute` are updated.

The patch has no performance effect measurable by any of the actual Class-File API benchmarks.

Please review.

Thanks,
Adam

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8335820](https://bugs.openjdk.org/browse/JDK-8335820): java/lang/invoke/LFCaching/LFSingleThreadCachingTest.java fails due to IllegalArgumentException: hash must be nonzero (**Bug** - P3)

### Reviewers
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20074/head:pull/20074` \
`$ git checkout pull/20074`

Update a local copy of the PR: \
`$ git checkout pull/20074` \
`$ git pull https://git.openjdk.org/jdk.git pull/20074/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20074`

View PR using the GUI difftool: \
`$ git pr show -t 20074`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20074.diff">https://git.openjdk.org/jdk/pull/20074.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20074#issuecomment-2214045252)

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/classfile/impl/BootstrapMethodEntryImpl.java [hunk 1]
```diff
@@ -76,9 +76,7 @@ public boolean equals(Object obj) {
 
     static int computeHashCode(MethodHandleEntryImpl handle,
                                List<? extends LoadableConstantEntry> arguments) {
-        int hash = handle.hashCode();
-        hash = 31 * hash + arguments.hashCode();
-        return AbstractPoolEntry.phiMix(hash);
+        return 31 * handle.hashCode() + arguments.hashCode();
```
Comment (liach) [inline src/java.base/share/classes/jdk/internal/classfile/impl/BootstrapMethodEntryImpl.java:79]: Should we update the algorithm here, as this algorithm processes hash codes of the 2nd last item of arguments and the handle the same, both just multiplied by 31?

Also need a copyright year update.

Comment (asotona) [reply src/java.base/share/classes/jdk/internal/classfile/impl/BootstrapMethodEntryImpl.java:79]: Right, BSMEntries are also stored in EntryMap.

**Review by liach — APPROVED** [review]

**Comment by liach** [comment]
@asotona This patch is ready for integration. You can still backport this critical fix before rdp1.

**Pull request closed without merging**